pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,6 +123,68 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .map(m -> {
+                               int indexOfSlash = reverseStringSearch('/', xpathExpressionArea.getText(), m.getInsertionEnd());
+                               String input = xpathExpressionArea.getText().substring(indexOfSlash + 1, m.getInsertionEnd());
+                               return Tuples.t(indexOfSlash, input);
+                           })
+                           .filter(t -> t._2.matches("[a-zA-Z]+"))
+                           .subscribe(e -> autoComplete(e._1, e._2));
+
+        xpathExpressionArea.addEventHandler(KeyEvent.KEY_PRESSED, t -> {
+            if (t.getCode().isLetterKey() || (t.isControlDown() && t.getCode().isWhitespaceKey()) || t.getCode() == KeyCode.BACK_SPACE) {
```

Comment:
The popup should be shown whenever you have some completion results to show, not when a letter in inserted. Remove that condition and add a show call to the end of the autocomplete method. Also, pressing ctrl+space should evaluate possible autocompletion results, starting before the search position. You may create an event stream for those key pressed and merge with the first to share logic